### DISCUSSION TITLE
Get "We are sorry... Invalid Request" when sending saml request to keycloak

### DISCUSSION THREAD

**Discussion by maivanhung12c1** [keycloak/keycloak#28538] [Q&A] [answered]
Hello everyone, 
Currently, I'm implementing saml2.0 and using keycloak for IDP 

I got "We are sorry... Invalid Request" when redirect to http://192.168.33.61:8080/realms/myrealm/protocol/saml?SAMLRequest=nVZZb6PKEn7Pr7A8jyhhtQ3WOEfsxgYMXsD4ZcTSLMasDcbm1x%2BSnCQzuXNH516Jh67qWr%2BvqsX3v27ZZXQFNUyKfDHGn7DxX88P33OIzdm2ifMtqFoAm9FglcP5oF6M2zqfFy5MBtHNAJw3%2FnzHauqceMLmZV00hV9cxh8O%2BJ8dXAhB3Qy5Pz2IxThumnKOol3XPXXkU1FHKIFhGIox6GAUwCT6Nh6x7558kcM2A%2FUO1NfEB4et%2BhEAZ4gnfEo%2FkeQTjtPzyUsQ14doWcBmPBKGxpLcbV47%2F0%2BPKT6nMRpDa%2BBeMohm99cD%2Bt4iCt1s6FMRFuMkeEwimdneDsLOVPEbGZbxcANhC5QcNm7eLMYERlCP2PAxe4ycU%2FScYE7jkfFPMC7JgySP%2FoyV92YE58v93ng0Nrv9a4BrEoBaH6wXYz6pS9dPf3TA%2B%2BEGWZL%2FeHEej6x3focw4%2BeBXXz%2BWl09koo6c5s%2F533RDC2Gr6ZzkDdJcx8%2F%2FwHiAaLb%2FccLPsQPb6gH5MHTQNx39DPxSxHEfJdEA%2F5tDUZKsBh%2FSPj48xoESh4WbzLv5kWe%2BO4l6V9Z00ATF8GIvURFnTRx9l8GB0dx7GVwHsHNf%2FRxKv82Rr%2Fk%2F5eRfhnBGrqPMHbx92BbEIIa5D4YHbbKYvztt1PxZrqv3Ry%2B4Am%2FyP9bBSC%2FgktRguARvjfyXsy%2Fj%2Fh7dNDflSkk0bAy%2Fw9WP%2BP0FsVyLy14xo5JUjA6F2CoNq2Tsw2DiaToU2GHLd5q%2BNn6TfOB8z%2Fy1yn5IPXNCcEtgTnjlzqNltT5jsxYpl2lZ7y59VBNjpgdMrVWSQWH6EFkVKt7fXBU0aZ1BvKT2t4%2FHGqFchpnv8mEaQxDyg7rFsu0zJ2aZaA760DqQxh602kSdx7WIB3MVsEGJSZhcCqNonrwNVSfrC4EeQHCsPVYyAZa3l6MTqrylKJpZ8eTAIXKLpKjtYbw7jmLdflmG5wj9jsdfbiafXvMxSxN64PunPEZJflREYf4gTsGHsn0k%2FRS6XfZi3PVb%2F1juJZS07tg5IG7eYUlPUx59WYJKVWmnn2T2aIMANI31MmVgsIPM0KstN1SOd5FbMPYahMFF9yhdHyqh7nZHwriwSr768ppr1TbFUlBb1Nf02J2sfhk4CfEX1lYg%2FsnJccJxghu435K%2FMvbHQ673IBnTVH4vOd51vF5LpI3La2vlD2rc1FaxWkiMx3GsSaUWIHda1uxEzpHsExTENj4HBBS6BNS4wkskDrsrgsspglRp%2FVap50ld9D1P%2BnumiC5GqfdxD1rcJFucaym8bJOOhbTOzYOh1o45fw1tyix7IZnTZp9ueej9XAWWfRwZsp9UQhB1TTkDLNyGTI67E%2BVf%2BcU0TSOclQRFVjF6JUTTydLMhwT7hEZQ3Fjk%2FlEiF2UbLtCNIvnhzdURpEV6dO5fjtpZz29Hbdbx%2FFIlQOGymIgWLd6Rl6rSaKbhU%2BDesJOrqqHlf4s3AZJOyFMRy7iJqYY7ry9EiLimLmk7XVviqyNlCO5WLZwy0lAzx5jqRIOEj9jhAlaXjJtq1vAJlQ6gAd%2Fh3AHlVA1QFUCyuh%2B4ISCdXR8oCNubdvs3iiCSm0RygT7Q0UQ9AlzzsY9g2qmMGsT9OnUOLG36%2FWq6EbsXGyqQpvmZOWbfNKosEUcBOyZqC%2B5aYbf61yayACKtWTbquCaisCaLKexmMzvKnmneKRgihzfHViWUgYuhDRIoFfToqUwyBKPG7eot%2Buts6mzmeskWdVcII3fE2t5DZX79WQ6Rt60Flzuo%2BHJ6v3AxbstI29paCSc5TJsOZX3YuJazlbBTyx3RVq5E%2BKNcCa3M5oHSwQPDxibV6fI85k0yqdn0qlc6zZZzRohstyZUOPajdwxsFQCEebr6SZgeHvXK8ZJ70raEIX8MiHdDdXcW1DZsTQNbswkW0qMs%2FINm09Uzz3oBsj4%2B9L0dY1kUhb4yTmWTah5OMV1y2G48x4s5V0XTgyqWZ2ifTJro1lpM44ooB1GTnac0yiY3KjOLNOWVTNRguUymvKcGGS6fdlZh2vAlUnpqUt5TcoFSCM95QkToZQss98W%2Betmfmrfthf9ZbN%2F3f1X%2Bde%2FxueHvwE%3D&RelayState=hwQFxJDni1OKb2iz&SigAlg=http%3A%2F%2Fwww.w3.org%2F2000%2F09%2Fxmldsig%23rsa-sha1&Signature=j0uong5NGX96ygJOgnev81ysMEk8UhlyWnev56IQeqb8xQ0V1foXS2tk68aBULGNkE0qjkhkA6A%2BJ71lX0xHpTMnkNIUHd03XbNDRa75BX7o8Btn3pzeXFHorJnhMaEDJeLcT30Za0Qk6HXIs8t%2F71lrcWhZltsHjh0PyY1%2Fb648wIHkcCqbMvF%2FfX%2BOQlXhhJNdZf%2FQJCMMAFZRdrFSCThfRNyGHn5XnpXmsC6B1tApkPHo7LCYNdJCyTc9afyk8oMpx0308ybijpCqHbiiXc7WACXD%2BzwEAX17Lxl6MMoYA6bUPljno3GFJgoju5FrnsqbmOe4H69%2BP%2B%2F3AvKd7w%3D%3D

Do you know what is wrong with my URL? How can I fix it?
Thank you for your help

**Comment by codespearhead** [comment]
Start by making sure that:
- System time on the servers are not skewed.
- The URL you're redirecting to after authentication is mentioned in the client's valid redirect URIs.
- The `Client ID` in Keycloak matches the `Entity ID` specified in the SAML request.

If none of the above work, we'll need to see what comes up in the Keycloak server's console after clicking that URL.

**Comment by maivanhung12c1** [comment]
![setting_saml](https://github.com/keycloak/keycloak/assets/48687271/591c2164-39f0-4e12-92ea-5446eaffbcc6)
This is my setup, I am running Keycloak as an Identity Provider (IdP) and a Service Provider (SP) on the same server to ensure the system time remains synchronized. Are my settings correct?

**Reply by codespearhead** [reply, marked as answer]
Follow this tutorial [1]. And here's a full-fledged example [2].

[1] https://docs.anchore.com/current/docs/configuration/user_authentication/sso/mapping/keycloak/
[2] https://github.com/PabloLec/sb_keycloak_saml

**Comment by maivanhung12c1** [comment]
Thanks a lot! Following the tutorial you suggested, I realized my mistake. The Client ID should be the Entity ID of the Service Provider (SP), not the Keycloak Identity Provider (IdP).

**Reply by codespearhead** [reply]
I'm happy to hear that!

Can you [mark my previous comment as the answer](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#marking-a-comment-as-an-answer) to this discussion then?

**Reply by maivanhung12c1** [reply]
Sure, I can do that

**Reply by prakashshnu10** [reply]
@maivanhung12c1 I am facing the same issue, can you help me?
